server/punishments.ts

```diff
@@ -1358,24 +1374,31 @@ export const Punishments = new class {
 	 * @return Descriptive text for the remaining time until the punishment expires, if any.
 	 */
 	checkLockExpiration(userid: string | null) {
-		if (!userid) return ``;
+		if (!userid) return '';
+
+		const user = Users.get(userid);
+		if (user?.permalocked) return '(expires never)';
+
 		const punishment = Punishments.userids.get(userid);
 
 		if (punishment) {
-			const user = Users.get(userid);
-			if (user?.permalocked) return ` (never expires; you are permalocked)`;
-			const expiresIn = new Date(punishment[2]).getTime() - Date.now();
-			const expiresDays = Math.round(expiresIn / 1000 / 60 / 60 / 24);
-			let expiresText = '';
-			if (expiresDays >= 1) {
-				expiresText = `in around ${Chat.count(expiresDays, "days")}`;
-			} else {
-				expiresText = `soon`;
-			}
-			if (expiresIn > 1) return ` (expires ${expiresText})`;
+			return ` (expires ${this.getPunishmentExpirationDescription(punishment)})`;
 		}
 
-		return ``;
+		return '';
+	}
+
+	getPunishmentExpirationDescription(punishment: Punishment) {
+		const expiresIn = new Date(punishment[2]).getTime() - Date.now();
+		const expiresDays = Math.round(expiresIn / 1000 / 60 / 60 / 24);
+		const expiresHours = Math.round(expiresIn / 1000 / 60 / 60);
+		if (expiresDays >= 1) {
+			return `in around ${Chat.count(expiresDays, "days")}`;
+		} else if (expiresHours >= 1) {
+			return `in around ${Chat.count(expiresHours, "hours")}`;
+		} else {
+			return 'within the next hour';
+		}
```

Comment:
Just use ``Chat.toDurationString`` for this.

Reply:
Sure, makes sense. Updated in the most recent version.